src/main/java/org/jboss/threads/EnhancedQueueExecutorBase3.java

```diff
@@ -48,6 +48,10 @@
      * Use a spin lock for the head lock.
      */
     static final boolean HEAD_SPIN = readBooleanPropertyPrefixed("head-spin", true);
+    /**
+     * Use synchronized for the head lock.
+     */
+    static final boolean HEAD_SYNCHRONIZED = readBooleanPropertyPrefixed("head-synchronized", false);
```

Comment:
Instead of adding additional properties, we could update the `head-lock` and `tail-lock` properties to be backwards compatible, but support `spin`, `synchronized`, and `reentrant` in additional to the existing boolean values.